src/syncer.cpp

```diff
@@ -245,34 +238,49 @@ void Syncer::upsyncLocalChanges(const QDateTime &,
     }
 
     // now upsync the changes for each addressbook
-    if (modifiedAddressbookUrls.size()) {
-        Q_FOREACH (const QString &addressbookUrl, modifiedAddressbookUrls) {
-            m_cardDav->upsyncUpdates(addressbookUrl,
-                                     added[addressbookUrl],
-                                     modified[addressbookUrl],
-                                     deleted[addressbookUrl]);
+    if (!m_syncError && !m_syncAborted) {
```

Comment:
I don't see anything in this function that could cause m_syncError or m_syncAborted to be set - so better to move this test to the beginning and avoid all the unused calculations.